c10/util/MaybeOwned.h

```diff
@@ -157,7 +160,6 @@ class MaybeOwned final {
         isBorrowed_ = false;
       }
     }
-    TORCH_INTERNAL_ASSERT_DEBUG_ONLY(isBorrowed_ == rhs.isBorrowed_);
```

Comment:
What's going on here?

Reply:
In a debug build, it may throw,  breaking the noexcept specification